browser/src/layer/marker/TextInput.js

```diff
@@ -1244,14 +1248,19 @@ L.TextInput = L.Layer.extend({
 			if (this._justSwitchedToEditMode && accept && this._isInitialContent()) {
 				// We need to make the paragraph at the cursor position focused in core
 				// so its content is sent to the editable area.
-				window.app.console.log('A11yTextInput._setAcceptInput: going to emit a synthetic click after switching to edit mode.');
 				this._justSwitchedToEditMode = false;
-				var top = this._map._docLayer._visibleCursor.getNorthWest();
-				var bottom = this._map._docLayer._visibleCursor.getSouthWest();
-				var center = L.latLng((top.lat + bottom.lat) / 2, top.lng);
-				var cursorPos = this._map._docLayer._latLngToTwips(center);
-				this._map._docLayer._postMouseEvent('buttondown', cursorPos.x, cursorPos.y, 1, 1, 0);
-				this._map._docLayer._postMouseEvent('buttonup', cursorPos.x, cursorPos.y, 1, 1, 0);
+				// A small delay for the blinking cursor to be updated
+				setTimeout(function() {
+					if (!this._map._docLayer || !this._map._docLayer._visibleCursor)
+						return;
+					window.app.console.log('A11yTextInput._setAcceptInput: going to emit a synthetic click after switching to edit mode.');
+					var top = this._map._docLayer._visibleCursor.getNorthWest();
+					var bottom = this._map._docLayer._visibleCursor.getSouthWest();
+					var center = L.latLng((top.lat + bottom.lat) / 2, top.lng);
+					var cursorPos = this._map._docLayer._latLngToTwips(center);
+					this._map._docLayer._postMouseEvent('buttondown', cursorPos.x, cursorPos.y, 1, 1, 0);
+					this._map._docLayer._postMouseEvent('buttonup', cursorPos.x, cursorPos.y, 1, 1, 0);
+				}.bind(this), 100);
```

Comment:
can we do that more deterministic?
that after required event we call this code

Reply:
I pushed a new version that doesn't need delay. I restricted the code to text document only.